x-pack/test/cases_api_integration/security_and_spaces/tests/common/attachments_framework/external_references.ts

```diff
@@ -0,0 +1,446 @@
+
+      expect(ref).to.eql({
+        id: 'my-id',
+        name: 'externalReferenceId',
+        type: 'test-type',
+      });
+
+      expect(comment).to.eql({
+        ...withoutId,
+        created_by: defaultUser,
+        pushed_at: null,
+        pushed_by: null,
+        updated_by: null,
+      });
+    });
+
+    it('should create references and attributes correctly if externalReferenceType===so on user actions', async () => {
+      const postedCase = await createCase(supertest, postCaseReq);
+      await createComment({
+        supertest,
+        caseId: postedCase.id,
+        params: postExternalReferenceSOReq,
+      });
+
+      const userActions = await getCaseUserActions({ supertest, caseID: postedCase.id });
+      const createCommentUserAction = userActions[1];
+
+      const esResponse = await getSOFromKibanaIndex({
+        es,
+        soType: CASE_USER_ACTION_SAVED_OBJECT,
+        soId: createCommentUserAction.action_id,
+      });
+
+      const commentOnES = esResponse.body._source?.[CASE_USER_ACTION_SAVED_OBJECT]?.payload.comment;
+      const ref = getReference(esResponse, postExternalReferenceSOReq.externalReferenceId);
+
+      const { externalReferenceId, ...withoutId } = postExternalReferenceSOReq;
+      expect(ref).to.eql({
+        id: 'my-id',
+        name: 'externalReferenceId',
+        type: 'test-type',
+      });
+
+      expect(commentOnES).to.eql(withoutId);
+    });
+
+    it('should create references and attributes correctly if externalReferenceType===doc', async () => {
+      const postedCase = await createCase(supertest, postCaseReq);
+      const patchedCase = await createComment({
+        supertest,
+        caseId: postedCase.id,
+        params: postExternalReferenceESReq,
+      });
+
+      const esResponse = await getSOFromKibanaIndex({
+        es,
+        soType: CASE_COMMENT_SAVED_OBJECT,
+        soId: patchedCase.comments![0].id,
+      });
+
+      const commentOnES = esResponse.body._source?.[CASE_COMMENT_SAVED_OBJECT];
+      const comment = removeServerGeneratedPropertiesFromSavedObject(commentOnES);
+      const ref = getReference(esResponse, postExternalReferenceSOReq.externalReferenceId);
+
+      expect(ref).to.be(undefined);
+      expect(comment).to.eql({
+        ...postExternalReferenceESReq,
+        created_by: defaultUser,
+        pushed_at: null,
+        pushed_by: null,
+        updated_by: null,
+      });
+    });
+
+    it('should create references and attributes correctly if externalReferenceType===doc on user actions', async () => {
+      const postedCase = await createCase(supertest, postCaseReq);
+      await createComment({
+        supertest,
+        caseId: postedCase.id,
+        params: postExternalReferenceESReq,
+      });
+
+      const userActions = await getCaseUserActions({ supertest, caseID: postedCase.id });
+      const createCommentUserAction = userActions[1];
```

Comment:
Instead of relying on the ordering, it'd probably be better to just filter the array based on type.